tools/Customization/DevHome.Customization/Views/FileExplorerPage.xaml.cs

```diff
@@ -44,16 +42,15 @@ public FileExplorerPage()
 
     public void AddButton_Click(object sender, RoutedEventArgs e)
     {
-        RootPath = RootPathTextBox.Text;
-
-        if (ValidateRepositoryPath(RootPath))
+        var numProviders = ViewModel.LocalRepositoryProviders.Count;
+        if (numProviders == 1)
```

Comment:
If DevHome has only 1 provider, register the repository.  Id Devhome has more than 1 repository, we show that repo in the flyout?  How does `ShowAt` add the repo?

Reply:
The flyout shoes the provider names if we have more than one. The flyout name can then be clicked on to call a function that registers the repository.

Reply:
I understand.  I mostly understand.  A button calls AddButton_Click.  The the number of providers is more than 1 they are shown to the user and the user will pick a provider.  On the click of the flyout that provider is registered.  Maybe add some comments explaining this behavior?  The method names confused me.